### ISSUE TITLE
Add text rendering

### ISSUE THREAD

**Issue by MeerkatBoss** [B05-232/plug-standard#47]
It is currently impossible to render any text using `plug::RenderTarget`. We need to address this issue sooner or later.

**Comment by Sigmarik** [comment]
It is possible to render it to a texture and then draw the texture to the render target.

But sure, it would be more covenient to render text directly to the target without having to deal with giant intermediate CPU-side textures.

**Comment by MeerkatBoss** [comment]
How can we render text to texture if we can't use SFML or SDL2 inside plugins? Don't they conflict with each other if used together (due to OpenGL being state machine)?

**Comment by Sigmarik** [comment]
We can, but in very limited capacity. They cannot have shared buffers, but each library can utilize their own rendering capabilities for internal drawing (like text rendering with plug::Texture export).

Otherwise I can't wrap my head around how we are going to do basic things like loading textures from file. Sure, we could each write our own image importer or use a library like stb_image, but I doubht everyone will be willing to do any of these.

**Comment by MeerkatBoss** [comment]
> We can, but in very limited capacity. They cannot have shared buffers, but each library can utilize their own rendering capabilities for internal drawing (like text rendering with plug::Texture export).

So we can actually use `sf::RenderTexture` in plugin, imported by SDL2 application? This would be great, if we had proof of it. I guess we'll have to test it as soon as we get any SDL2 implementation.

> Otherwise I can't wrap my head around how we are going to do basic things like loading textures from file. Sure, we could each write our own image importer or use a library like stb_image, but I doubht everyone will be willing to do any of these.

IIRC, SFML does not use textures (and by extent any OpenGL capabilities) when loading `sf::Image`, so it should not conflict with SDL2. However, this should be tested as well.

**Comment by Sigmarik** [comment]
Yes, texture loading should not interact with the GPU. But we are dealing with SFML here, so I would not be so certain.

Text rendering should not be a problem as long as SFML does not turn the OpenGL state machine into a mess.
